renderdoc/driver/vulkan/vk_pixelhistory.cpp

```diff
@@ -2427,6 +2555,55 @@ bool VulkanDebugManager::PixelHistorySetupResources(PixelHistoryResources &resou
   vkr = m_pDriver->vkCreateImageView(m_Device, &viewInfo, NULL, &stencilImageView);
   RDCASSERTEQUAL(vkr, VK_SUCCESS);
 
+  if(samples != VK_SAMPLE_COUNT_1_BIT)
+  {
+    vkr = m_pDriver->vkBindImageMemory(m_Device, stagingImage, gpuMem, stagingImageOffset);
+    RDCASSERTEQUAL(vkr, VK_SUCCESS);
+
+    viewInfo.image = stagingImage;
+    viewInfo.viewType = VK_IMAGE_VIEW_TYPE_2D_ARRAY;
+    viewInfo.subresourceRange = {VK_IMAGE_ASPECT_COLOR_BIT, 0, VK_REMAINING_MIP_LEVELS, 0,
+                                 VK_REMAINING_ARRAY_LAYERS};
+
+    uint32_t bs = GetByteSize(1, 1, 1, format, 0);
+
+    if(bs == 1)
+      viewInfo.format = VK_FORMAT_R8_UINT;
+    else if(bs == 2)
+      viewInfo.format = VK_FORMAT_R16_UINT;
+    else if(bs == 4)
+      viewInfo.format = VK_FORMAT_R32_UINT;
+    else if(bs == 8)
+      viewInfo.format = VK_FORMAT_R32G32_UINT;
+    else if(bs == 16)
+      viewInfo.format = VK_FORMAT_R32G32B32A32_UINT;
+
+    if(viewInfo.format == VK_FORMAT_UNDEFINED)
+    {
+      RDCERR("Can't copy 2D to Array with format %s", ToStr(format).c_str());
+    }
+
+    vkr = m_pDriver->vkCreateImageView(m_Device, &viewInfo, NULL, &stagingImageView);
+    RDCASSERTEQUAL(vkr, VK_SUCCESS);
+
+    viewInfo.image = targetImage;
+    viewInfo.subresourceRange = {VK_IMAGE_ASPECT_COLOR_BIT, sub.mip, 1, sub.slice, 1};
```

Comment:
The image covers the specific slice, so when we do a copy later we choose slice 0.